tuplex/codegen/src/TypeAnnotatorVisitor.cc

```diff
@@ -1190,6 +1194,14 @@ namespace tuplex {
     }
 
     void TypeAnnotatorVisitor::assignHelper(NIdentifier *id, python::Type type) {
+        if(_ongoingLoopCount != 0 && !_loopTypeChange) {
```

Comment:
a bit lost on this comparison, can we add a comment on what is checked here? I.e., what the intuition is?

Reply:
It is detecting any type change inside loops. _ongoingLoopCount indicates that the assign happens in some loop body. If _loopTypeChange == True then some type change already occurs. Added in comments.